File: CW_1/bas_processor/domain/bank_account.py
class BankAccount:
    def __init__(self, name: str, balance: float = 0.0):
        self.name = name
        self.balance = balance

    def deposit(self, amount: float):
        if amount < 0:
            raise ValueError("Deposit amount cannot be negative")
        self.balance += amount

    def withdraw(self, amount: float):
        if amount < 0:
            raise ValueError("Withdraw amount cannot be negative")
        if amount > self.balance:
            raise ValueError("Insufficient funds")
        self.balance -= amount

File: CW_1_CSharp/Domain/BankAccount.cs
using System;

namespace CW_1_CSharp.Domain
{
    public class BankAccount
    {
        private int _id;
        private string _name;
        private double _balance;

        public BankAccount(int id, string name, double balance)
        {
            _id = id;
            if (string.IsNullOrWhiteSpace(name))
                throw new ArgumentException("Название счёта не может быть пустым!");
            _name = name;
            _balance = balance;
        }

        public int Id => _id;

        public string Name
        {
            get => _name;
            set
            {
                if (string.IsNullOrWhiteSpace(value))
                    throw new ArgumentException("Название счёта не может быть пустым!");
                _name = value;
            }
        }

        public double Balance => _balance;

        public void UpdateBalance(double amount)
        {
            _balance += amount;
        }

        public override string ToString()
        {
            return $"[BankAccount #{_id}] {_name} (balance: {_balance})";
        }
    }
}